src/gui/sdl_mapper.cpp

```diff
@@ -2101,7 +2097,12 @@ static void SetActiveEvent(CEvent * event) {
 			SetActiveBind(*(mapper.abindit));
 
 			const auto new_bind = dynamic_cast<CKeyBind*>(*mapper.abindit);
-			drop_other_bound_events(new_bind);
+			// TODO: version of drop_other_bound_events that works
+			// for joystick binds, for which above line assigns
+			// nullptr to newbind and causes assertion failure
```

Comment:
Are you going to address this TODO as part of this PR?

Reply:
AFAICT this would be non-trivial to fix (may involve significant refactoring of class hierarchy), and IMO is out of scope for this PR. The argument could be made that I should have left this broken and not bothered commenting; a fix could be done separately, which would be more maintainable. However, while this was broken it was difficult to test the hotplug support because I could not even click on a joystick event in the mapper UI without a crash. The change in this PR preserves @interloper98 's dropping of other bound events for keyboard keys - the only thing it worked on - while removing the crash when this function is called for joystick buttons/axes/etc.

I would suggest @interloper98 take care of properly generalizing the `drop_other_bound_events` function if they are available, as they authored this section I needed to fix.

Reply:
Thanks for the TODO, and yeah, probably out of scope for this PR, indeed.